services/src/main/scala/cromwell/services/womtool/models/MetaValueElementJsonSupport.scala

```diff
@@ -0,0 +1,31 @@
+package cromwell.services.womtool.models
+
+import cats.syntax.functor._
+//import io.circe.{Encoder, Json}
+import io.circe.{ Decoder, Encoder }, io.circe.generic.auto._
+import io.circe.syntax._
+
+import wom.callable.MetaValueElement
+import wom.callable.MetaValueElement._
+
+object MetaValueElementJsonSupport {
+
+  implicit val metaValueElementEncoder: Encoder[MetaValueElement] = Encoder.instance {
+    case MetaValueElementNull => throw new Exception("null not supported yet")
+    case a @ MetaValueElementBoolean(_) => a.asJson
+    case a @ MetaValueElementFloat(_) => a.asJson
+    case a @ MetaValueElementInteger(_) => a.asJson
+    case a @ MetaValueElementString(_) => a.asJson
+    case MetaValueElementObject(_) => throw new Exception("object not supported yet")
+    case MetaValueElementArray(_) => throw new Exception("array not supported yet")
+  }
+
+  implicit val metaValueElementDecoder: Decoder[MetaValueElement] =
+    List[Decoder[MetaValueElement]](
+//      Decoder[MetaValueElementNull].widen,
```

Comment:
Is this the part you're having trouble with? Your message about needing help refers both to this file, and to `WorkflowDescription.scala`.

Reply:
Right, it is this file. I understand how to use `spray.json`, more or less, but this is `io.circe`, which behaves differently. It is a bit mysterious to me.

Reply:
Before the vacation, I was trying to rewrite this decoder/encoder like this: https://github.com/dnanexus-rnd/cromwell/blob/orodeh_meta_section_tng/services/src/main/scala/cromwell/services/womtool/models/MetaValueElementJsonSupport.scala. I don't know if that is better.

Reply:
I don't think you actually need to implement the decoder, we don't do anything with it - see `cromwell.services.womtool.models.WorkflowDescription#workflowDescriptionDecoder` which only decodes some basic stuff for a test

Reply:
The `typeName` stuff in `WomTypeJsonSupport` is because we are describing the type of an input, not its value - I don't think that would be necessary for the meta section elements.

Reply:
This is very helpful. Let's see if I can make the code work now.